packages/atomic/src/utils/store.ts

```diff
@@ -2,21 +2,44 @@ import {
   NumericFacetValue,
   DateFacetValue,
   SortCriterion,
+  Facet,
+  CategoryFacet,
+  NumericFacet,
+  DateFacet,
+  SearchEngine,
 } from '@coveo/headless';
 import {VNode} from '@stencil/core';
 import {ObservableMap} from '@stencil/store';
+import {DependencyCondition} from '../components/facets/facet-depends-on/facet-dependency-conditions';
+import {FacetWithType} from '../components/facets/facet-depends-on/facet-union-types';
 
-interface FacetInfo {
+interface FacetInfo<
+  F extends Facet | CategoryFacet | NumericFacet | DateFacet
+> {
   label: string;
+  facetId: string;
+  facet?: F;
 }
 
 interface FacetValueFormat<ValueType> {
   format(facetValue: ValueType): string;
   content?(facetValue: ValueType): VNode;
 }
 
+export interface FacetDependency {
+  parentFacetId: string;
+  dependantFacetId: string;
```

Comment:
Ah, that's because when I searched them, I thought `dependant` was a noun, but it looks like that's specific to british English. I'll use `dependent` across the board, it's easier.